src/crypto/x509/x509.go

```diff
@@ -278,7 +280,8 @@ var publicKeyAlgoName = [...]string{
 	DSA:          "DSA",
 	ECDSA:        "ECDSA",
 	Ed25519:      "Ed25519",
-	EdDilithium3: "Ed25519-Dilithium3",
+	EdDilithium2: "Ed25519-Dilithium2",
+	EdDilithium3: "Ed448-Dilithium3",
```

Comment:
nits: should these be named differently, like `Ed25519Dilithium2` and `Ed448Dilithium3`?

Reply:
I don't want to break current users. Let's reconsider naming when we add `Ed25519-ML-DSA-44`.